src/libs/SelectionScraper/index.js

```diff
@@ -80,6 +81,16 @@ const getHTMLOfSelection = () => {
         }
     }
 
+    // Get the div housing the UnreadActionIndicator and remove it from copied content.
+    // As it currently stands, 'accessibilityHints.newMessageLineIndicator' is only used inside
+    // UnreadActionIndicator.js which is also only used in ReportActionItem.js, so this is a relatively
+    // safe move.
+    const newMessageLineIndicatorDiv = div.querySelector(`[aria-label="${Localize.translateLocal('accessibilityHints.newMessageLineIndicator')}"]`);
```

Comment:
Have we considered adding a `className` to the `UnreadActionIndicator` & selecting that here? This way we could have 1 specific class that's used for all elements that need the "disable selection" feature, and we can select it here & remove them all at once?

Reply:
I believe that's what we tried in the [previous PR that led to a regression](https://github.com/Expensify/App/pull/18228) using the `user-select: none` class, @Beamanator .

Given we're using React Native, styles are applied using the `style` prop, so I haven't come across a scenario like when building web apps where a non-styling class is applied to an element and that class is then used to query the DOM.

It's, for this reason, [I suggested using an attribute](https://github.com/Expensify/App/pull/18228#issuecomment-1534797205) earlier as my second option, but we all consented to dealing with just this issue now as we continue to look for a more robust solution to the numerous copy/pasting issues. 

In addition, React Native's `View` and `Text` components are limited in what styles they accept in the `style` prop, so I feel like applying an arbitrary class to them that does nothing e.g. `borderWidth: 0` which we would then use to query the DOM to remove unselected elements would be counterintuitive.

If you inspect the Expensify website, you'll see that the `div`s don't have a `class` attribute with a meaningful name - they all look something like this. 

```HTML
<div 
    class="css-175oi2r" 
    style="position: absolute; top: -10px; left: 0px; width: 100%; height: 20px; padding-right: 20px; padding-left: 20px; flex-direction: row; align-items: center; z-index: 1; cursor: default;"
>
//...
</div>
```

The element's styling is applied inline by the `style` attribute.

With all this considered, I settled on using the `aria-label` that was unique to `UnreadActionIndicator`.

Reply:
> I believe that's what we tried in the [previous PR that led to a regression](https://github.com/Expensify/App/pull/18228) using the `user-select: none` class, @Beamanator .

Not exactly - in the last PR we just looked for dom elements with a specific style, and that style was used in multiple places. I'm saying we should have a class and query elements based on the unique class name.

> Given we're using React Native, styles are applied using the `style` prop, so I haven't come across a scenario like when building web apps where a non-styling class is applied to an element and that class is then used to query the DOM.

Aah I think you're right that React Native doesn't support `className` anyway 🤔  But we can use `nativeID` maybe which I see us using in lots of other places in the app!

> In addition, React Native's `View` and `Text` components are limited in what styles they accept in the `style` prop, so I feel like applying an arbitrary class to them that does nothing e.g. `borderWidth: 0` which we would then use to query the DOM to remove unselected elements would be counterintuitive.

Just to be clear I think you're talking about an arbitrary style `attribute`, not `class`

> If you inspect the Expensify website, you'll see that the `div`s don't have a `class` attribute with a meaningful name - they all look something like this.

Right, this is auto-generated somewhere 👍 

@fedirjh what do you think about trying to use `nativeID` so we can have a unique identifier for this component, instead of using an `aria-label` selector? I imagine that specific text in the aria-label won't be used in other components in the future but we never know, and i feel like a unique `nativeID` is more likely to never be duplicated

Reply:
Yes, I was talking about a style attribute in React Native (this would be a CSS property on the web). 

I actually considered using `nativeID` and even the `id` prop supplied to `View`s because these were unique., but I didn't see any `div` with such an attribute.  

The big question would be how to get that `id` value to show on the web so we can use it if at all such an attribute is supplied to a `View` or `Text`. 

Circling back to your initial suggestion of giving a class to all elements that shouldn't be selected, should IDs be used, since they're unique, every time it's decided an element shouldn't be selected, the DOM will have to be queried using a different ID each time to get all the elements i.e. you can't do `div.querySelectorAll("${elementID}")`.

Reply:
@fedirjh I honestly wish I saw that line sooner and knew `nativeID` would be converted to `id` in the HTML.

As a matter of fact, that's actually what I considered in my very first proposal, but wasn't sure `nativeID` would work this way. 

One thing I'd like to suggest though. 

Since `style.userSelectNone` is already being applied to `UnreadActionIndicator`, wouldn't the following code be redundant?

Also, since each `id` needs to be unique, I don't think using `div.querySelectorAll('#no-copy-area')` would work here because only one element would be queried from the DOM.

```javascript
#no-copy-area {
    user-select: none;
    -moz-user-select: none;
    -ms-user-select: none;
    -webkit-user-select: none;
}

// use it like
nativeID="no-copy-area"

// query it
div.querySelectorAll(`#no-copy-area`).forEach(el => el.remove())
```

Reply:
By the way, I've just given `UnreadActionIndicator` a `nativeID` prop and adjusted `SelectionScraper` to query an element by this ID locally, and it's working all right. 

No console errors, and some of the tests I've done so far (the ones in Tests in the PR description) are passing.  

Would you like me to go through all the tests and commit the changes to this branch, @Beamanator once everything checks out?

Reply:
First of all, I love the idea of putting the ID name in a CONST 👍 

Second, at Expensify we try not to do too much future-proofing because we often consider this "over-engineering". If we did this everywhere, we believe we would waste a lot of time coming up with solutions to problems that we don't even know if they're real problems.

In this issue / proposal / fix in general we've done a lot of thinking into the future, so it's maybe a bit hard to back out now 😅  but in my opinion option 1 from your investigation is perfect enough. In the future IF someone else wants to use the same solution, they can do some refactoring to get your 2nd solution.

But for now I think option 1 sounds perfecto 👍 

Thoughts @fedirjh ?

Reply:
No worries, @Beamanator. 

I'll eat my food one bite at a time going forward.

Reply:
Thanks for understanding @Victor-Nyagudi 👍  And thanks for talking it through too, it's always good to chat about these things before diving too deep 👍